tsl/src/compression/compress_utils.c

```diff
@@ -528,6 +543,14 @@ tsl_decompress_chunk(PG_FUNCTION_ARGS)
 
 	if (uncompressed_chunk->relkind == RELKIND_FOREIGN_TABLE)
 	{
+		/*
+		 * Updating the chunk compression status of the Access Node BEFORE executing remote
+		 * decompression. In the event of failure, the compressed status will be cleared. The
+		 * distributed compression policy will attempt to compress again, which is idempotent, thus
+		 * the metadata are eventually consistent.
+		 */
+		ts_chunk_set_compressed_chunk(uncompressed_chunk, INVALID_CHUNK_ID, false);
```

Comment:
The explanation should be in the comment but it's probably not clear.

The distributed compression policy depends on this field. In the event of failure:
1. if we set it BEFORE, then the policy will think it's uncompressed, try to compress it again, and since it's idempotent it should be ok.
2. if we set it AFTER, then the policy will think it's compressed, will never compress it again, but it's really uncompressed in practice.

This analysis is based on the decision to not involve 2PC in this.